config/electrum.go

```diff
@@ -0,0 +1,98 @@
+package config
+
+import (
+	"embed"
+	"fmt"
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+	"github.com/keep-network/keep-core/pkg/bitcoin/electrum"
+	"math/rand"
+	neturl "net/url"
+	"strings"
+)
+
+//go:embed _electrum_urls/*
+var electrumURLs embed.FS
+
+// readElectrumConfigs reads Electrum configs from an embedded file for the
+// given Bitcoin network.
+func readElectrumConfigs(network bitcoin.Network) ([]electrum.Config, error) {
```

Comment:
`*Configs` or `*Config`? Even if we had multiple entries in the embedded file, we'd still be reading "a configuration" for Electrum connection.

Reply:
I think the current version makes sense as this function returns a list of possible configurations. Then `resolveElectrum` picks up one randomly.

Reply:
I see. I got confused about what is actually returned. Why don't we rename `readElectrumConfigs` to `readElectrumUrls` and change its returned type? See that we only return host + protocol, and later in `resolveElectrum` we ignore all other settings that would potentially come in the config returned by `readElectrumConfigs`:

https://github.com/keep-network/keep-core/blob/51127b93140c795409116b42948f5fe0dad6f181/config/electrum.go#L92-L95

Side note (cc @nkuba - maybe we can fix it as a part of the work on the `wss://` support?), our config for `[bitcoin.Electrum]` is not consistent with the config for `[ethereum]`.

Compare:

https://github.com/keep-network/keep-core/blob/51127b93140c795409116b42948f5fe0dad6f181/configs/config.toml.SAMPLE#L3-L4

https://github.com/keep-network/keep-core/blob/51127b93140c795409116b42948f5fe0dad6f181/configs/config.toml.SAMPLE#L49-L56

Scheme (`wss://`,`ssl://`, `tcp://`) should [be a part of the URL](https://www.rfc-editor.org/rfc/rfc1738#section-5) just like we have it for `[ethereum]`. If we want to resolve the protocol later, this should be an auxiliary function on the `Config` type parsing the protocol from the URL and not a separate field:

https://github.com/keep-network/keep-core/blob/51127b93140c795409116b42948f5fe0dad6f181/pkg/bitcoin/electrum/config.go#L27-L28

If we do this change, then we can return just an URL from `readElectrumUrls` without bothering about tuples or introducing other types.

Reply:
> Why don't we rename readElectrumConfigs to readElectrumUrls and change its returned type? See that we only return host + protocol, and later in resolveElectrum we ignore all other settings that would potentially come in the config returned by readElectrumConfigs:

That was my first approach but it was better to return `[]electrum.Config` than an array of tuples (note that we need to parse `electrum.Protocol`). However, if we plan to merge `electrum.URL` with `electrum.Protocol` in the config, here is the intermediary solution: cf266a69

Reply:
Captured it in https://github.com/keep-network/keep-core/issues/3580. I will keep this thread open for reference.